### PR TITLE
optimize experience on SteamOS (Steam Deck)

### PR DESCRIPTION
**Pull request by LoyieKing**
- bugfix: chinese version game can not launch on SteamOS gaming mode (see 80ffbba697392399b3f4a410464930003eefcb25)
- feat: settings page: add close button & esc/alt-f4 to exit. steamos gaming mode will hide windows' title bar, cause settings window can not exit (see 1c92e533b425437cc8a0262e9193914e59557b01)
- feat: if runnning on SteamOS gaming mode, all windows will be fullscreen
- coding: add neutralino.js type definition, remove all `decalre const Neutralino:any`  (see 6bbd85e7fe55672e0e43aafa9e128059caec0686)

### CODE REVIEW

File: src/index.svelte [hunk 1]
```diff
@@ -53,7 +55,7 @@
         constants.paths.launcherDir.then((launcherDir) => {
             Neutralino.filesystem.getStats(`${launcherDir}/logs/latest.log`)
                 .then(async () => {
-                    let created_at = (await Neutralino.os.execCommand(`stat -c '%W' "${path.addSlashes(`${launcherDir}/logs/latest.log`)}"`)).stdOut;
+                    let created_at = Number.parseInt((await Neutralino.os.execCommand(`stat -c '%W' "${path.addSlashes(`${launcherDir}/logs/latest.log`)}"`)).stdOut);
```
Comment (Mar0xy) [inline src/index.svelte:58]: Wouldn't this break created_at as it uses the full date format as seen on line 59

File: src/ts/Voice.ts [hunk 2]
```diff
@@ -105,8 +105,8 @@ export default class Voice
 
                     // Parse installed voice packages
                     Neutralino.filesystem.readDirectory(voiceDir)
-                        .then(async (files) => {
-                            files = files.filter((file) => file.type == 'DIRECTORY')
+                        .then(async (filesString) => {
+                            const files = filesString.filter((file) => file.type == 'DIRECTORY')
```
Comment (Mar0xy) [inline src/ts/Voice.ts:109]: Why are you changing all "files" variables to "filesString" when there is no real need as you just add overhead when then proceeding to make a new const for "files"

File: src/ts/core/HDiffPatch.ts [hunk 3]
```diff
@@ -10,14 +10,14 @@ export default class HDiffPatch
         return new Promise(async (resolve) => {
             let result = await Neutralino.os.execCommand(`./public/hdiffpatch/hpatchz -f "${path.addSlashes(file)}" "${path.addSlashes(patch)}" "${path.addSlashes(output)}"`);
 
-            result = (result.stdOut ?? result.stdErr).includes('patch ok!');
+            const ret = (result.stdOut ?? result.stdErr).includes('patch ok!');
```
Comment (Mar0xy) [inline src/ts/core/HDiffPatch.ts:13]: normally result doesn't get cut short to "ret" but "res" instead as "ret" normally stands for return

File: src/settings.svelte [hunk 4]
```diff
@@ -215,6 +224,12 @@
                 >{$_(`settings.${item}.title`)}</div>
             {/each}
         </div>
+        
+        
+        <div class="save-button" >
+            <!-- svelte-ignore missing-declaration -->
+            <button class="button" on:click={()=>Neutralino.events.dispatch("windowClose")} >✔️</button>
+        </div>
```
Comment (Mar0xy) [inline src/settings.svelte:232]: Please provide screenshots when adding new UI elements to so we can determine if they fit in or not

File: src/splash.svelte [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <script context="module" lang="ts">
-    declare const Neutralino;
+    
 </script>
```
Comment (Mar0xy) [inline src/splash.svelte:3]: You are removing the declare yet leaving the empty script element in each file

File: src/ts/FPSUnlock.ts [hunk 6]
```diff
@@ -2,7 +2,7 @@ import { Downloader, path } from '../empathize';
 
 import constants from './Constants';
 
-declare const Neutralino;
+
```
Comment (Mar0xy) [inline src/ts/FPSUnlock.ts:5]: Same for all .ts files you are leaving an unnecessary extra line

File: src/ts/Voice.ts (hunk 2 above)
Comment (Mar0xy) [inline src/ts/Voice.ts:109]: This adds overhead and shouldn't be changed.

File: src/index.svelte (hunk 1 above)
Comment (LoyieKing) [reply src/index.svelte:58]: > Wouldn't this break created_at as it uses the full date format as seen on line 59

- `stat %W` always outputs integer
- `Number.parseInt` returns NaN if input is not a number, and never throw error
- see L61, L63, original code directly assert that `created_at` is a number, so the current way is undoubtedly safer.

so it's fine

**Comment by krypt0nn** [comment]
Merged, some parts will be rewritten in later commits

**Pull request merged**